### ISSUE TITLE
Cannot undo changes using vim emulation in vscode

### ISSUE THREAD

**Issue by imkhali** [VSCodeVim/Vim#8258]
Type: <b>Bug</b>

Somehow when I make different edits in a file, and if I try to undo them, I got the "already at the oldest change" in the status bar after one or no undo at all, hence, I cannot undo the changes anymore. I have been using vim with vscode for a while and it has never happened before. I guess but not sure if it is related to the latest vscode upgrade.

Extension version: 1.24.3
VS Code version: Code 1.75.1 (441438abd1ac652551dbe4d408dfcec8a499b8bf, 2023-02-08T21:32:34.589Z)
OS version: Windows_NT x64 10.0.19044
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz (4 x 3408)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.89GB (6.98GB free)|
|Process Argv|--crash-reporter-id 248c4a3d-013b-4abf-bc5a-0b2205438698|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30669237
azure-dev_surveyone:30548225
vsccc:30610678
pyindex848:30662994
nodejswelcome1cf:30587006
3biah626:30602489
pyind779cf:30662993
f6dab269:30613381
pythonsymbol12:30657548
pythonb192:30669360

```

</details>

**Comment by FateScript** [comment]
The same issue here:
VSCode Info: 
```
Version: 1.75.1
Commit: 441438abd1ac652551dbe4d408dfcec8a499b8bf
Date: 2023-02-08T21:34:59.000Z (1 wk ago)
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Darwin x64 19.6.0
Sandboxed: Yes
```
vscodevim.vim version 1.24.3

This issue might be caused by update of VSCode, I never update my vim simulator. May it helps.

**Comment by unta1337** [comment]
Same here.
And it sometimes results in the wrong state as below.

1. Type 'i' to get insert mode.
2. Type 'pizza'
3. Hit esc to enter normal mode.
4. Type 'i' again and enter ' is great'.
5. Press 'u' two times to undo twice.
6. Expected to left with empty string, but 'p' is still remaining.

And frustratingly I don't konw how to reproduce this bug. It feels random at all.
Since undo function has a problem, my code sometimes corrupted because of vscode vim.

As @imkhali mentioned, it seems it has to do with recent vscode update.
In code-server this problem doesn't seen to be occured.
(Note: latest version of code-server is 4.10)

[vscode]
Version: 1.75.1 (user setup)
Commit: 441438abd1ac652551dbe4d408dfcec8a499b8bf
Date: 2023-02-08T21:32:34.589Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Windows_NT x64 10.0.19045
Sandboxed: No

[code-server]
code-server: v4.9.1
Code: 1.73.1
Commit: 6261075646f055b99068d3688932416f2346dd3b
Date: 2022-12-07T20:58:35.136Z
Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36 Edg/110.0.1587.49

**Comment by Komar0ff** [comment]
Hi! This is a known problem. https://github.com/VSCodeVim/Vim/issues/8157

**Comment by FateScript** [comment]
Comment [here](https://github.com/VSCodeVim/Vim/issues/8157#issuecomment-1438314999) solve my issue.

**Comment by J-Fields** [comment]
Duplicate of https://github.com/VSCodeVim/Vim/issues/8157, which should be fixed in `v1.25.1`
